dora/core/client/fs/src/main/java/alluxio/client/file/dora/DoraCachePositionReader.java

```diff
@@ -51,17 +54,20 @@ public int readInternal(long position, ReadTargetBuffer buffer, int length)
     if (position >= mFileLength) { // at end of file
       return -1;
     }
+    int originalOffset = buffer.offset();
     try {
       return mNettyReader.read(position, buffer, length);
     } catch (PartialReadException e) {
       int bytesRead = e.getBytesRead();
       if (bytesRead == 0) {
-        // we didn't make any progress, throw the exception so that the caller needs to handle that
-        Throwables.propagateIfPossible(e.getCause(), IOException.class);
-        throw new IOException(e.getCause());
+        LOG.debug("Failed to read file from worker through Netty", e);
+        return fallback(position, buffer, length, originalOffset);
```

Comment:
check `PropertyKey.DORA_CLIENT_UFS_FALLBACK_ENABLED` before proceeding to fall back to UFS.

Reply:
any case that we do not want user to fallback to ufs? 
the case that client does not have ufs credential?

Reply:
No ufs credentials is one case. Most often due to buggy code the client should be able to read from worker but it doesn't. Ufs fallback hides the underlying error and provides an illusion that everything is working out fine. Then you have to debug poor performance with no real, obvious errors.